process/receiptslog/receiptsManager.go

```diff
@@ -17,17 +27,26 @@ type ArgsGenerateReceiptsAndSave struct {
 }
 
 type receiptsManager struct {
-	trieInteractor Interactor
+	trieInteractor     Interactor
+	receiptsDataSyncer ReceiptsDataSyncer
+	marshaller         marshal.Marshalizer
+	hasher             hashing.Hasher
 }
 
 // NewReceiptsManager will create a new instance of receipts manager
 func NewReceiptsManager(args ArgsReceiptsManager) (*receiptsManager, error) {
 	if check.IfNil(args.TrieHandler) {
 		return nil, ErrNilTrieInteractor
 	}
+	if args.ReceiptsDataSyncer == nil {
```

Comment:
Use `check.IfNil()`. Also, add a check for the marshaller and hasher

Reply:
done